### ISSUE TITLE
docker-compose build does not work with -f path under certain conditions

### ISSUE THREAD

**Issue by andreivasiletruta** [docker/compose#1064]
Have a docker-compose installation
Have a docker-compose.yml file in path /A/B/
CD to /A
Trigger: ```docker-compose -f /A/B/docker-compose.yml build```

Result: 

``````
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 31, in main
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 21, in sys_dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 27, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 24, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 59, in perform_command
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 119, in build
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.project", line 167, in build
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 480, in build
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 429, in build
TypeError: You must specify a directory to build in path

``````

**Comment by dnephin** [comment]
This error is actually about the contents of your `docker-compose.yml`, not the path to it.  Note that paths in `docker-compose.yml` are relative to the current working directory.

**Comment by thaJeztah** [comment]
@andreivasiletruta were you able to solve your issue with the comment above?

**Comment by cristianmedeiros** [comment]
@thaJeztah I do, I had "build: ./fig/proxy" on my yml file but ./fig wasn't there.

**Comment by kevana** [comment]
@dnephin Are the docs incorrect here?

>build
Path to a directory containing a Dockerfile. When the value supplied is a relative path, it is interpreted as relative to the location of the yml file itself.

**Comment by thaJeztah** [comment]
@dnephin I think this issue can be closed https://github.com/docker/compose/issues/1064#issuecomment-88089349

**Comment by dnephin** [comment]
Ya, i think this would have been fixed in 1.2
